File: Problem_Solving/Algorithms/Warmup/6_Plus_Minus/Solution.cs
using System.CodeDom.Compiler;
using System.Collections.Generic;
using System.Collections;
using System.ComponentModel;
using System.Diagnostics.CodeAnalysis;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.Serialization;
using System.Text.RegularExpressions;
using System.Text;
using System;

class Solution {

    static void plusMinus(int[] arr) {
        double posNums=0;
        double negNums=0;
        double zeroNums=0;
        double posFraction=0.0;
        double negFraction=0.0;
        double zeroFraction=0.0;
        int arrLen=arr.Length;

        for(int i=0;i<arrLen;i++){
            if(arr[i]==0)
                zeroNums++;
            else if(arr[i]>0)
                posNums++;
            else
                negNums++;
        }

        posFraction=posNums/arrLen;
        negFraction=negNums/arrLen;
        zeroFraction=zeroNums/arrLen;

        Console.WriteLine(posFraction.ToString("0.000000") + "\n" + 
            negFraction.ToString("0.000000") + "\n" + 
            zeroFraction.ToString("0.000000"));
    }

    static void Main(string[] args) {
        int n = Convert.ToInt32(Console.ReadLine());

        int[] arr = Array.ConvertAll(Console.ReadLine().Split(' '), arrTemp => Convert.ToInt32(arrTemp));
        plusMinus(arr);
    }
}

File: Problem_Solving/Algorithms/Warmup/6_Plus_Minus/Solution.php
<?php

function plusMinus($arr) {
    $posNums=0.0;
    $negNums=0.0;
    $zeroNums=0.0;
    $posFraction=0.0;
    $negFraction=0.0;
    $zeroFraction=0.0;
    $arrLen=count($arr);

    for($i=0;$i<$arrLen;$i++){
        if($arr[$i]===0)
            $zeroNums++;
        else if($arr[$i]>0)
            $posNums++;
        else
            $negNums++;
    }

    $posFraction=$posNums/$arrLen;
    $negFraction=$negNums/$arrLen;
    $zeroFraction=$zeroNums/$arrLen;
    
    printf("%.6f\n%.6f\n%.6f",$posFraction,$negFraction,$zeroFraction); 
}

$stdin = fopen("php://stdin", "r");

fscanf($stdin, "%d\n", $n);

fscanf($stdin, "%[^\n]", $arr_temp);

$arr = array_map('intval', preg_split('/ /', $arr_temp, -1, PREG_SPLIT_NO_EMPTY));

plusMinus($arr);

fclose($stdin);

?>
